src/System.Private.CoreLib/shared/System/String.Comparison.cs

```diff
@@ -594,8 +594,13 @@ public bool EndsWith(string value, bool ignoreCase, CultureInfo culture)
 
         public bool EndsWith(char value)
         {
-            int thisLen = Length;
-            return thisLen != 0 && this[thisLen - 1] == value;
+            int lastPos = Length - 1;
+            if ((uint)lastPos < (uint)Length && this[lastPos] == value)
+            {
+                return true;
+            }
+
+            return false;
```

Comment:
Please add a comment linking to the JIT issue about `if (x) return true; else return false;`... the proliferation of this workaround bugs me :smile:

Reply:
Its a habit I didn't realise I had till @mikedn pointed it out. 

The asm returning the bool directly looks better https://github.com/dotnet/coreclr/pull/22206#issuecomment-457606417 and its still better than previous even if the Jit currently then has issues merging the compares when it inlines into an if; so am returning the result directly now